tests/catch/unit/memory/hipHostRegister.cc

```diff
@@ -156,5 +152,49 @@ TEMPLATE_TEST_CASE("Unit_hipHostRegister_Memcpy", "",
 
   free(A);
   free(Bh);
-  hipFree(Bd);
+  HIP_CHECK(hipFree(Bd));
+}
+
+template <typename T> __global__ void fill_kernel(T* dataPtr, T value) {
+  size_t tid{blockIdx.x * blockDim.x + threadIdx.x};
+  dataPtr[tid] = value;
+}
+
+TEMPLATE_TEST_CASE("Unit_hipHostRegister_Negative", "", int, float, double) {
+  TestType* hostPtr = nullptr;
+
+  size_t sizeBytes = 1 * sizeof(TestType);
+  SECTION("hipHostRegister Negative Test - nullptr") {
+    HIP_CHECK_ERROR(hipHostRegister(hostPtr, 1, 0), hipErrorInvalidValue);
+  }
+
+  hostPtr = reinterpret_cast<TestType*>(malloc(sizeBytes));
+  SECTION("hipHostRegister Negative Test - zero size") {
+    HIP_CHECK_ERROR(hipHostRegister(hostPtr, 0, 0), hipErrorInvalidValue);
+  }
+
+#if HT_NVIDIA
```

Comment:
Did you file a ticket? We should return an error too

Reply:
I think the assumption here was that flags are completely ignored on AMD implementation and as such it didn't make sense to write negative testing for it. 

If that's not the case, then there is more errors that could be returned to match Nvidia implementation. Only 0x00 | 0x04 | 0x02 | 0x01 | 0x08 should be considered valid values for flags. 

We can add that extra negative testing and open a ticket to return error if the flags don't match those values. Would that make sense?

Reply:
sure